utilities/roslz4/src/lz4s.c

```diff
@@ -613,7 +616,10 @@ int roslz4_buffToBuffDecompress(char *input, unsigned int input_size,
 
   int ret;
   ret = roslz4_decompressStart(&stream);
-  if (ret != ROSLZ4_OK) { return ret; }
+  if (ret != ROSLZ4_OK) {
+    roslz4_decompressEnd(&stream);
```

Comment:
Same idea here, though in this case, I don't think that `roslz4_decompressStart` can leak memory on an allocation failure, so we might not want to change this at all.

Reply:
I agree with you that this shouldn't be necessary since `roslz4_decompressStart` only allocates when returning `OK`. So in case of it failing it doesn't need to deallocate anything.